src/post.js

```diff
@@ -10,12 +10,42 @@ module.exports = async config => {
   const name = `v${pkg.version}`;
   const tag = {owner, repo, ref: `refs/tags/${name}`, sha: await promisify(gitHead)()};
   const body = await promisify(plugins.generateNotes)(config);
-  const release = {owner, repo, tag_name: name, name, target_commitish: branch, draft: !!debug, body};
+  let githubPreRelease = false;
+  let release = {
+    owner,
+    repo,
+    tag_name: name,
+    name,
+    target_commitish: branch,
+    draft: !!debug,
+    body,
+    prerelease: githubPreRelease,
+  };
 
   if (debug && !githubToken) {
     return {published: false, release};
   }
 
+  // obtain npm dist tag if any
+  let npmDistTag = (pkg.publishConfig || {}).tag;
+
+  // if npmDistTag is set, we can consider github pre release
+  if (npmDistTag != null) {
```

Comment:
As determined in [src/index.js#L54](https://github.com/semantic-release/semantic-release/blob/caribou/src/index.js#L54) `config.npm.tag` cannot be `null`;

Reply:
How do I check whether it was manually set or not in this case?

Reply:
Maybe I'm missing, but what difference does it make if the user explicitly set the dist-tag as `latest` in `publishConfig` or in the code in `index.js` set it as it's default value of `latest` ?
If it's `latest` then no Github pre-release, independently of how it was set isn't it?

Reply:
Thought you mentioned that it shouldn't pre-release when set to 'latest' **by default**.

Reply:
I might have express myself in a confusing way, sorry.
The algorithm I think should be something along those line:
- if dist-tag (default or not) is `latest` or in `no-prerelease` property => no pre-release
- Otherwise => pre-release